TransIT.API/Controllers/IssueController.cs

```diff
@@ -81,29 +49,93 @@ public override async Task<IActionResult> Get([FromQuery] uint offset = 0, uint
             }
         }
 
+        [HttpGet("{id}")]
+        public virtual async Task<IActionResult> Get(int id)
+        {
+            var result = await _serviceFactory.IssueService.GetAsync(id);
+            return result != null
+                ? Json(result)
+                : (IActionResult)BadRequest();
+        }
+
+        [HttpGet("/search")]
+        public virtual async Task<IActionResult> Get([FromQuery] string search)
+        {
+            var result = await _serviceFactory.IssueService.SearchAsync(search);
+            return result != null
+                ? Json(result)
+                : (IActionResult)BadRequest();
+        }
+
         [HttpPost]
-        public override async Task<IActionResult> Create([FromBody] IssueDTO obj)
+        public async Task<IActionResult> Create([FromBody] IssueDTO obj)
         {
-            IActionResult result = await base.Create(obj);
+            var createdEntity = await _serviceFactory.IssueService.CreateAsync(obj);
             await _issueHub.Clients.Group(RoleNames.Engineer).SendAsync("ReceiveIssues");
-            return result;
+            return createdEntity != null
+                ? CreatedAtAction(nameof(Create), createdEntity)
+                : (IActionResult)BadRequest();
+        }
+
+        [HttpPut("{id}")]
+        public async Task<IActionResult> Update(int id, [FromBody] IssueDTO obj)
+        {
+            obj.Id = id;
+
+            var result = await _serviceFactory.IssueService.UpdateAsync(obj);
+            return result != null
+                ? NoContent()
+                : (IActionResult)BadRequest();
+        }
+
+        [HttpDelete("{id}")]
+        [Authorize(Roles = "ADMIN")]
+        public async Task<IActionResult> Delete(int id)
+        {
+            await _serviceFactory.IssueService.DeleteAsync(id);
+            return NoContent();
+        }
+
+        [DataTableFilterExceptionFilter]
+        [HttpPost("~/api/v1/datatable/[controller]")]
+        public override async Task<IActionResult> Filter(DataTableRequestDTO model)
+        {
+            var isCustomer = User.FindFirst(RoleNames.Schema)?.Value == RoleNames.Register;
+            var userId = User.FindFirst(ClaimTypes.NameIdentifier).Value;
```

Comment:
please check with Andriy because we may have the current user in UserService as this user is set in Action filter, if yes use it everywhere it's possible